scripts/text_classification/fasttext_word_ngram.py

```diff
@@ -218,31 +241,95 @@ def get_context(args):
     return context
 
 
+def get_length(inp):
+    """Returns the length"""
+    return float(len(inp[0]))
+
+
+def get_sequence(inpt):
+    """Transforms input to vocab id's"""
+    document = inpt[0]
+    vocab = inpt[1]
+    return vocab[document.split()]
+
+
+def convert_to_sequences(dataset, vocab):
+    """This function takes a dataset and converts
+    it into sequences via multiprocessing
+    """
+    start = time.time()
+    dataset_vocab = map(lambda x: (x, vocab), dataset)
+    with mp.Pool() as pool:
+        # Each sample is processed in an asynchronous manner.
+        output = pool.map(get_sequence, dataset_vocab)
```

Comment:
Have you compared the time taken without multiprocessing? Serialization overhead can be very large (and increase significantly with increasing problem size, ie. larger vocab).

Reply:
yeah, the script ran with dbpedia dataset itself with n-gram=2 pretty fast.
This change made the runs faster compared to the slowdown here problem that existed earlier.

Reply:
How does the vocabulary size scale with ngram=x for dbpedia? Do you have some concrete numbers for the convert_to_sequences function? Doing only vocab[document.split()] on the separate process I'm concerned the serialization overhead will be higher than the runtime of splitting and looking up.

Reply:
I see, i didnt run any concrete numbers per say, my earlier code didnt have mp, but upon adding mp, i havent noticed any slow down. Infact, it now processes every dataset much faster